### PR TITLE
Insights & Perso

### PR DESCRIPTION
**Pull request by aseure**
# Insights

```scala
client.execute {
  send event ClickedFilters("eventName", "indexName", []string{"filter"})
  send event ClickedObjectIDs("eventName", "indexName", []string{"objectID"})
  send event ClickedObjectIDsAfterSearch("eventName", "indexName", []string{"objectID"}, []string{42} /* positions */, "queryID")

  // Conversion events
  send event ConvertedObjectIDs("eventName", "indexName", []string{"objectID"})
  send event ConvertedObjectIDsAfterSearch("eventName", "indexName", []string{"objectID"}, "queryID")
  send event ConvertedFilters("eventName", "indexName", []string{"filter"})

  // View events
  send event ViewedFilters("eventName", "indexName", []string{"filter"})
  send event ViewedObjectIDs("eventName", "indexName", []string{"objectID"})

  // Send raw events
  send event InsightsEvent(...)
  send events Seq(InsightsEvent(...), ...)
}
```

# Personalization

```scala
client.execute {
  // Retrieve personalization strategy
  get strategy

  // Set personalization strategy
  set strategy Strategy(...)
}
```

### CODE REVIEW

File: src/main/scala/algolia/inputs/InsightsEvent.scala [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+import algolia.AlgoliaUtils
+import org.json4s.CustomSerializer
+
+case class InsightsEvent(
+    eventType: String,
+    eventName: String,
+    index: String,
+    userToken: String,
+    timestamp: Long = AlgoliaUtils.now(),
```
Comment (Ant-hem) [inline src/main/scala/algolia/inputs/InsightsEvent.scala]: Can't we set it as `Option[Long] = None` ?

Comment (aseure) [reply src/main/scala/algolia/inputs/InsightsEvent.scala]: Right thanks 👍 Completely forgot it in Scala

Comment (aseure) [reply src/main/scala/algolia/inputs/InsightsEvent.scala]: Actually, it will simply the tests 🎉

**Comment by aseure** [comment]
All review changes amended @Ant-hem

**Pull request merged**